### ISSUE TITLE
Bug related to Definition Lists and normal Lists

### ISSUE THREAD

**Issue by facelessuser** [Python-Markdown/markdown#1005]
Source:

```py
import markdown

md = """
Apple
:   Pomaceous fruit of plants of the genus Malus in
    the family Rosaceae.

    - paragraph

        paragraph
"""


output = markdown.markdown(md, extensions=['def_list'])
print(output)
```

Output:

```html
<dl>
<dt>Apple</dt>
<dd>
<p>Pomaceous fruit of plants of the genus Malus in
the family Rosaceae.</p>
<ul>
<li>paragraph</li>
</ul>
<pre><code>paragraph
</code></pre>
</dd>
</dl>
```

Related to #1004 

I will most likely solve this as a separate issue from #1004 as the extension is constructed a bit differently, even if the issue manifests the same. The reason is most likely the same, but the path there is not exactly the same.

**Comment by waylan** [comment]
Could this be a duplicate of #918? Or is this different due to the inclusion of Admonitions?

**Comment by facelessuser** [comment]
Yeah this may be a duplicate. It looks similar. I'll close as a dupe if it turns out to be the same. I only glanced over it. I plan on digging into this over the weekend. I think I've already solved the admonition variant locally.

**Comment by facelessuser** [comment]
This is a duplicate of the other. I think I have a grasp on what is going on and how to fix it now.

**Comment by facelessuser** [comment]
Closing as a duplicate and will fix the issue #918.
